StepperEngine/src/Flow/Flow.java

```diff
@@ -276,19 +281,43 @@ public void execution(ArrayList<DataType> inputs){
         try {
             for (Step step : steps) {
                 step.execute();
-                if (step.getStatus() == Step.Status.Failure)
+                setFlowStatus(step);
```

Comment:
Why not just setStatus(Status.Fail)? like why is the function "SetFlowStatus(Step)" needed?